### PR TITLE
feat: improve Dockerfile and include basic usage documentation

### PR DESCRIPTION
**Pull request by TrooperT**
I found this project while looking into self-hosting tailscale and would like to contribute as I can to help a great fledgling piece of software.

- Improves Dockerfile by using a multi-stage build sequence to improve layer caching.
- Pins against go@1.18 and alpine@3.15
- Volumes support
- Default command
- Basic Documentation and Usage in Markdown format

### CODE REVIEW

**Comment by jsiebens** [comment]
hi @TrooperT, thanks for your interest and appreciate the help but I'm a little bit torn an accepting this PR. 

I'm using [goreleaser](https://goreleaser.com/) to build and "release" binaries and docker images of this project. Because of that, a multi stage docker build is not really necessary (see [Docker Images](https://goreleaser.com/customization/docker/))
New preview versions are regularly pushed to ghcr.io/jsiebens/ionscale, so you should be able to run it with eg:

```
docker run \
    -v $(pwd)/conf:/data/conf \
    -v $(pwd)/db:/data/db \
    ghcr.io/jsiebens/ionscale:v0.0.1-preview7 server --config /data/config/ionscale.yaml
```
But I would not recommended it (yet), it's still in alpha/preview :-) 
(Also: not sure why added a policies volume, that's not required for ionscale)

I'm aware of the lack of documentation and I hope to provide some soon

**Comment by TrooperT** [comment]
Ah. I'm not familiar with goreleaser and might just look into it.

The policies volume was a vestige of my learning/exploration curve on setting tailnet IAM policies from within the container itself with ionscale prior to realizing I could do so via RPC from outside

TL;DR oops forgot to remove something.

Definitely interested in the way this project is heading with its support for multiple tailnets and already functional OIDC authentication.

I'll close this one out.
